### PR TITLE
Format all python code with black and isort

### PR DESCRIPTION
**Pull request by justinchuby**
**Description**

- Format all python code with black and isort
- Add format checks in CI
- Skip additional generated files in isort

**Motivation and Context**

Standardize formatting

Fixes https://github.com/onnx/onnx/issues/4278

### CODE REVIEW

**Comment by jcwchen** [comment]
Reopen to trigger release CIs

File: tools/style.sh
Comment (jcwchen) [inline tools/style.sh:25]: So after this PR, isort and black check will be "required" in ONNX CIs. I am good with this and I just want to mention that we need to reflect this in the document. I will take care of this by https://github.com/onnx/onnx/pull/4332.

**Review by jcwchen — COMMENTED** [review]
Overall looks good to me. Only one nit comment. Thanks for the formatting!

File: onnx/__init__.py
Comment (justinchuby) [reply onnx/__init__.py:3]: A sorted import list will cause import errors (onnx partially initialized). To avoid errors like this, torch.onnx requires developers to only import modules: https://github.com/pytorch/pytorch/wiki/torch.onnx-Namespacing#naming-and-usage-convention

**Review by jcwchen — APPROVED** [review]
LGTM. Please get approval from @onnx/sig-operators-approvers as well. Since it's a huge change, I prefer to forward other PRs which involve Python and are close to merge first to save duplicate work of resolving conflict. For instance, https://github.com/onnx/onnx/pull/4350 and https://github.com/onnx/onnx/pull/4351.

**Comment by gramalingam** [comment]
Why did the generated documentation files change?

**Comment by gramalingam** [comment]
What format check was added to CI (as mentioned in the PR description)? I couldn't find any. Thanks!

**Comment by justinchuby** [comment]
@gramalingam Doc changed because they contain source (test etc) that are formatted. black and isort were added (tools/style.sh)

**Comment by justinchuby** [comment]
@jcwchen I think it is easier to resolve conflicts the other way around, since git is very good at recognizing where small functions changes should go (in the formatted files). But if we wait for those changes to go in first, the whole reformatting process needs to be run again, essentially the work of creating a new PR.

**Comment by jcwchen** [comment]
> @jcwchen I think it is easier to resolve conflicts the other way around, since git is very good at recognizing where small functions changes should go (in the formatted files). But if we wait for those changes to go in first, the whole reformatting process needs to be run again, essentially the work of creating a new PR.

I remembered I often resolved code conflict about Python import before when merging PRs, but perhaps git is good at pure import reordering. Then I am fine with either way. If we decide to forward this PR first, then I will suggest we forward this ASAP, because this PR brings more required code check and we should accomplish it before more new PRs are proposed to prevent confusion (i.e., previously their PRs pass required CI, but after merging this PR they fails).

**Comment by justinchuby** [comment]
Let us know if this looks good to you @gramalingam ? If so we can move forward with this pr and coordinate with the ones waiting

**Comment by justinchuby** [comment]
Closed in favor of #4427

**Pull request closed without merging**
